engine/classes/ElggEntity.php

```diff
@@ -1220,12 +1220,17 @@ public function canAnnotate($user_guid = 0, $annotation_name = '') {
 			$return = false;
 		}
 
+		$hooks = _elgg_services()->hooks;
+
 		$params = array(
 			'entity' => $this,
 			'user' => $user,
 			'annotation_name' => $annotation_name,
 		);
-		return _elgg_services()->hooks->trigger('permissions_check:annotate', $this->type, $params, $return);
+		$return = $hooks->trigger("permissions_check:annotate:$annotation_name", $this->type, $params, $return);
```

Comment:
I think it's a bug that creating annotations named `""` works, so I will change this as you suggest.